rosbag2_transport/test/rosbag2_transport/test_keyboard_controls.cpp

```diff
@@ -135,24 +135,20 @@ TEST_F(RosBag2PlayTestFixture, test_keyboard_controls)
   auto prepared_mock_reader = std::make_unique<MockSequentialReader>();
   prepared_mock_reader->prepare(messages, topics_and_types);
   auto reader = std::make_unique<rosbag2_cpp::Reader>(std::move(prepared_mock_reader));
-  auto keyboard_handler = new MockKeyboardHandler(
-    play_options_.pause_resume_toggle_key,
-    play_options_.play_next_key);
+  auto keyboard_handler = std::make_shared<MockKeyboardHandler>();
 
   // play bag in thread
   auto player = std::make_shared<CountPlayer>(
-    std::move(reader), std::shared_ptr<KeyboardHandler>(keyboard_handler),
+    std::move(reader), keyboard_handler,
     storage_options_, play_options_);
   std::thread player_thread = std::thread([player]() {player->play();});
 
   EXPECT_THAT(player->is_paused(), false);
-  keyboard_handler->play_next();
-  EXPECT_THAT(player->is_paused(), false);
-  keyboard_handler->pause_resume();
+  keyboard_handler->simulate_key_press(play_options_.pause_resume_toggle_key);
```

Comment:
@emersonknapp Thanks for fixing tests.

However I think it would better to call 
```cpp
  EXPECT_THAT(player->is_paused(), false);
  keyboard_handler->simulate_key_press(play_options_.pause_resume_toggle_key);
  EXPECT_THAT(player->is_paused(), true);
```
before 
```cpp
std::thread player_thread = std::thread([player]() {player->play();});
```
to avoid race conditions and flakiness in tests.